ravenpy/utilities/data_assimilation.py

```diff
@@ -191,3 +194,173 @@ def update_state(x, qobs_pert, qobs_error, qsim):
     xa = np.maximum(xa, 0)
 
     return xa
+
+
+def perturb_full_series(model, std, start_date, end_date, dists, n_members=25):
+    """
+    This function will take a timeseries as input and perturb each variable
+    according to the desired uncertainty/confidence level and distribution of
+    each variable. Returns an Xarray Dataset object with n_members x time values
+    for each variable.
+
+    Parameters
+    ----------
+    model : Raven model class object
+        The model that will be used to perform the simulations and assimilation
+    std : tuple
+        Pairs of variable: standard deviation of the uncertainty.
```

Comment:
How does this work if dist is not norm ? There is always just one parameter ?
Maybe a dict would be a better type here than a tuple.

Reply:
it seems it is already a dictionnary, I just mixed up the terms because I didn't define it using "dict(...)" but it seems I am still learning :)